From 10a19dea64ead8fa18aab4d1fb047c43112d1409 Mon Sep 17 00:00:00 2001
From: TungDang <tung.dang.jy@rvc.renesas.com>
Date: Sun, 21 Aug 2016 07:24:34 +0700
Subject: [PATCH 6/9] waylandsink: Don't wait for the wl_surface_frame callback
 to notify the compositor of a wl_buffer

When the frame rendering delays, the weston compositor can drop frames
and catch up with the on-time processing by itself.

Porting from a patch in Linux 3.10 environment made by Kazunori Kobayashi <kkobayas@igel.co.jp>

Signed-off-by: TungDang <tung.dang.jy@rvc.renesas.com>
---
 ext/wayland/gstwaylandsink.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index 211bb6a..30064eb 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -785,7 +785,6 @@ gst_wayland_sink_render (GstBaseSink * bsink, GstBuffer * buffer)
   GstBuffer *to_render;
   GstWlBuffer *wlbuffer;
   GstFlowReturn ret = GST_FLOW_OK;
-  gint redraw_flag;

   g_mutex_lock (&sink->render_lock);

@@ -825,13 +824,6 @@ gst_wayland_sink_render (GstBaseSink * bsink, GstBuffer * buffer)

   wl_display_dispatch_queue_pending (sink->display->display, sink->frame_queue);

-  /* drop buffers until we get a frame callback */
-  redraw_flag = g_atomic_int_get (&sink->redraw_pending);
-  while (redraw_flag == TRUE) {
-    wl_display_dispatch_queue (sink->display->display, sink->frame_queue);
-    redraw_flag = g_atomic_int_get (&sink->redraw_pending);
-  }
-
   if (G_UNLIKELY (sink->video_info_changed)) {
     gst_wl_window_set_video_info (sink->window, &sink->video_info);
     sink->video_info_changed = FALSE;
--
1.9.1
